sync-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseFactory.java

```
@@ -36,30 +36,47 @@
+    public static SQLDatabase createSQLDatabase(String dbFilename, KeyProvider provider) throws IOException {
+
+        boolean runningOnAndroid =  Misc.isRunningOnAndroid();
+        boolean useSqlCipher = (provider.getEncryptionKey() != null);
 
         makeSureFileExists(dbFilename);
-        if(Misc.isRunningOnAndroid()) {
-            try {
-                Class c = Class.forName("com.cloudant.sync.sqlite.android.AndroidSQLite");
 
-                Method m = c.getMethod("createAndroidSQLite",String.class);
-                return (SQLDatabase)m.invoke(null,dbFilename);
+        try {
 
-            } catch (Exception e) {
-                logger.log(Level.SEVERE, "Failed to load database module", e);
-                return null;
-            }
-        } else {
-            try {
-                Class c = Class.forName("com.cloudant.sync.sqlite.sqlite4java.SQLiteWrapper");
-                Method m = c.getMethod("openSQLiteWrapper", String.class);
-                return (SQLDatabase)m.invoke(null, dbFilename);
-
-            } catch (Exception e) {
-                logger.log(Level.SEVERE,"Failed to load database module",e);
-                return null;
+            if (runningOnAndroid) {
+                if (useSqlCipher) {
+                    return (SQLDatabase) Class.forName("com.cloudant.sync.sqlite.android.AndroidSQLCipherSQLite")
+                            .getMethod("createAndroidSQLite", String.class, KeyProvider.class)
+                            .invoke(null, new Object[]{dbFilename, provider});
+                } else {
+                    return (SQLDatabase) Class.forName("com.cloudant.sync.sqlite.android.AndroidSQLite")
+                            .getMethod("createAndroidSQLite", String.class)
+                            .invoke(null, dbFilename);
+                }
+            } else {
+                if (useSqlCipher) {
+                    throw new IOException("No SQLCipher-based database implementation for Java SE");
```

is this really an `IOException`? i know that is the type in the `throws` clause but maybe it should be extended or we should throw an unchecked exception.